app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -1138,6 +1243,224 @@ private int getMoveID(Move move) {
+            cursor.moveToFirst();
+            while (!cursor.isAfterLast()) {
+                int uniqueID = cursor.getInt(cursor.getColumnIndex(POKEMON_UNIQUE_ID));
+                filteredUniqueids.add(uniqueID);
+                cursor.moveToNext();
+            }
+            cursor.close();
+
+        }
+        return filteredUniqueids;
+
+    }
+
+    /**
+     * Filters all unique ID's that are of the type requested
+     *
+     * @param typeID of the type in question
+     * @return array list of ints corresponding to unique ids
+     */
+    public ArrayList<Integer> getAllUniqueIDsFromType(int typeID){
+
+        ArrayList<Integer> filteredUniqueids = new ArrayList<>();
+        ArrayList<Integer> filteredNationalids = new ArrayList<>();
+
+        if(typeID > 0) {
+            String[] selectionArg = {String.valueOf(typeID)};
+            Cursor cursor = database.query(POKEMON_TYPES_TABLE, null, TYPE_ID + "=?", selectionArg, null, null, null);
+            cursor.moveToFirst();
+            while (!cursor.isAfterLast()) {
+                int uniqueID = cursor.getInt(cursor.getColumnIndex(POKEMON_UNIQUE_ID));
+                filteredUniqueids.add(uniqueID);
+                cursor.moveToNext();
+            }
+            cursor.close();
+
+        }
+        return filteredUniqueids;
+
+    }
+
+    /**
+     * Filters all unique ID's that are of the bewteen the range of stats or the stat requested
+     * @param lowerLimit lower integer value limit
+     * @param upperLimit upper integer value
+     * @param stat the stat being filtered for, where stat is either: HP, ATTACK, DEFENCE, spattack, spdefence or SPEED
+     *             TODO stat can also be SUM, which will search for SUM
+     * @return array of integers of the unique id's that correspond
+     */
+    public ArrayList<Integer> getAllUniqueIDsFromStat(int lowerLimit, int upperLimit, String stat){
+        //TODO: Add validation step for stat string
+        ArrayList<Integer> filteredUniqueids = new ArrayList<>();
+
+        String[] selectionArg = {String.valueOf(lowerLimit), String.valueOf(upperLimit)};
+        if(!stat.equals("SUM")) {
+            Cursor cursor = database.query(POKEMON_UNIQUE_INFO_TABLE, null, stat + ">=? AND " + stat + "<=?", selectionArg, null, null, null);
+            cursor.moveToFirst();
+            while (!cursor.isAfterLast()) {
+                int uniqueID = cursor.getInt(cursor.getColumnIndex(POKEMON_UNIQUE_ID));
+                filteredUniqueids.add(uniqueID);
+                cursor.moveToNext();
+            }
+            cursor.close();
+        }
+//        else{
+//            // Filter by the total sum of base stats
+//            // using a raw query to get sum, not sure if even possible with query
+//            Cursor cursor = database.rawQuery("select (HP + ATTACK + DEFENCE + SPATTACK + SPDEFENCE + SPEED) as sum from pokemon_unique_info where sum >="+String.valueOf(lowerLimit)+" AND sum <="+String.valueOf(upperLimit)+
```

Comment:
We should probably precalculate and place the Stat Sum in the DB. This should avoid the runtime calculation.